clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -279,6 +279,13 @@ public void abort() {
         aborted = true;
     }
 
+    public void unsetProducerState() {
```

Comment:
Hmm.. This doesn't actually unset the producer state. Maybe `reopen` would be a better name?